src/state/operator/selectors.js

```diff
@@ -33,11 +33,55 @@ import {
 } from '../locales/selectors'
 import { getGroups, makeLocaleGroupToken, getDefaultGroup } from '../groups/selectors'
 
-export const STATUS_AVAILABLE = 'available';
+import { STATUS_AVAILABLE, STATUS_RESERVE } from './constants';
 
 const percentAvailable = ( { load, capacity } ) => ( capacity - defaultTo( 0, load ) ) / capacity
 const totalAvailable = ( { load, capacity } ) => ( capacity - defaultTo( 0, load ) )
-const compare = ( a, b ) => {
+
+// This is the maximum number of chats a reserve operator can have before
+// another reserve operator is brought in to handle further chats. When
+// all reserve operators reach this limit, new chats will be balanced
+// between reserve operators.
+const RESERVE_MAX_CHATS = 2;
+
+/**
+/* Compare function for sorting operator priority. 
+ *
+ * The first operator in the resulting list should be the next one
+ * to be assigned to a chat. This is used for balancing chat load
+ * evenly across operators.
+ *
+ * @param {Operator} a First operator to compare
+ * @param {Operator} b Second operator to compare
+ * @return {Number} -1 if operator a should come first
+ *                  0 if both operators are equal priority
+ *                  1 if operator b should come first
+ */
+const compareOperatorPriority = ( a, b ) => {
+	// When comparing two operators, always prioritise the one whose
+	// status is available over the one who is in reserve
+	if ( a.status === STATUS_RESERVE && b.status === STATUS_AVAILABLE ) {
+		return 1;
+	}
+	if ( a.status === STATUS_AVAILABLE && b.status === STATUS_RESERVE ) {
+		return -1;
+	}
+
+	// When comparing two operators in reserve, prioritise the one
+	// already chatting to avoid disturbing the other.
+	if ( a.status === STATUS_RESERVE && b.status === STATUS_RESERVE ) {
+		const aLoad = a.load || 0;
+		const bLoad = b.load || 0;
+		if ( aLoad > 0 && aLoad < RESERVE_MAX_CHATS && bLoad === 0 ) {
```

Comment:
It may be, but if it is then something unexpected is going on, so we should probably not make any assumptions about what order to sort them in.